### PR TITLE
In 140593605 safety card

### PR DESCRIPTION
**Pull request by aaronOSI**

### CODE REVIEW

File: app/assets/javascripts/components/screenings/ScreeningEditPage.jsx [hunk 1]
```diff
@@ -112,12 +112,14 @@ export class ScreeningEditPage extends React.Component {
...
 
   mergeScreeningWithEdits(changes) {
-    const crossReportEdits = changes.get('cross_reports')
-    if (crossReportEdits) {
-      return this.state.screening.set('cross_reports', crossReportEdits).mergeDeep(changes)
-    } else {
-      return this.state.screening.mergeDeep(changes)
-    }
+    // Lists require to be reassigned the edits instead of appending.
+    const lists = changes.filter((val) => val.constructor.name === 'List')
```
Comment (NateOSI) [inline app/assets/javascripts/components/screenings/ScreeningEditPage.jsx]: I'm not sure what this function is doing? 
is there a test for it? 
also is there a better way of checking if a value is an Immutable List, i.e. Immutable.isCollection()?
https://facebook.github.io/immutable-js/docs/#/isCollection

File: app/assets/javascripts/utils/screeningEditHelper.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+export function saveParticipant(participant) {
+  return this.props.actions.saveParticipant(participant.toJS())
+    .then(() => {
+      this.props.actions.fetchScreening(this.props.params.id)
+    })
+}
+
+export function mergeScreeningWithEdits(changes) {
+  // Lists require to be reassigned the edits instead of appending.
```
Comment (saraOSI) [inline app/assets/javascripts/utils/screeningEditHelper.js]: This comment is new and confusing.

File: spec/features/screening/edit_screening_spec.rb [hunk 3]
```diff
@@ -40,13 +40,16 @@
...
-      .and_return(json_body(existing_screening.to_json, status: 200))
-
+      .and_return(body: existing_screening.to_json,
+                  status: 200,
+                  headers: { 'Content-Type' => 'application/json' })
     visit edit_screening_path(id: existing_screening.id)
-    expect(page).to have_content 'Edit Screening #My Bad!'
+  end
 
+  scenario 'edit an existing screening' do
```
Comment (saraOSI) [inline spec/features/screening/edit_screening_spec.rb]: Did we lose an expectation in the rearranging of this test? I don't see `expect(page).to have_content 'Edit Screening #My Bad!'`.

File: spec/features/screening/show_screening_spec.rb [hunk 4]
```diff
@@ -77,7 +77,11 @@
       expect(page).to have_css('th', text: 'Allegation(s)')
     end
 
-    expect(page).to have_css('#worker-safety-card.show', text: 'WORKER SAFETY')
+    within '#worker-safety-card', text: 'WORKER SAFETY' do
```
Comment (saraOSI) [inline spec/features/screening/show_screening_spec.rb]: There aren't any tests here to insure that the edit functionality works from the show page. In the past, I think we've tended to test the edge cases in the edit specs, but have at least one feature that does editing from the show page.

File: spec/features/screening/edit_screening_spec.rb [hunk 5]
```diff
@@ -242,6 +265,24 @@
     end
   end
 
+  scenario 'Worker safety card saves in isolation' do
```
Comment (saraOSI) [inline spec/features/screening/edit_screening_spec.rb]: Scenario for negative testing? What happens if the user presses cancel instead of save?

File: app/assets/javascripts/utils/screeningEditHelper.js [hunk 6]
```diff
@@ -0,0 +1,84 @@
+/* eslint-disable no-invalid-this */
```
Comment (NateOSI) [inline app/assets/javascripts/utils/screeningEditHelper.js]: These functions worry me a little as there are multiple dependencies this file library assumes are available
this.setState
this.props.screening
this.props.participants
this.state.screeningEdits
this.props.actions

If we want to pull these functions out they  would need to be pure functions and what ever dependancies should be passed through parameters.

File: spec/javascripts/components/screenings/ScreeningEditPageSpec.jsx [hunk 7]
```diff
@@ -77,6 +78,10 @@ describe('ScreeningEditPage', () => {
             {agency_type: 'District attorney', agency_name: 'SAC DA'},
             {agency_type: 'Licensing', agency_name: ''},
           ],
+          safety_alerts: [
```
Comment (saraOSI) [inline spec/javascripts/components/screenings/ScreeningEditPageSpec.jsx]: Does the safety_alerts array need to be both in `requiredProps` AND `requiredScreeningAttributes`?

File: spec/javascripts/components/screenings/ScreeningEditPageSpec.jsx [hunk 8]
```diff
@@ -620,6 +642,7 @@ describe('ScreeningEditPage', () => {
           id: '3',
           allegations: [],
           cross_reports: [],
+          safety_alerts: [],
```
Comment (saraOSI) [inline spec/javascripts/components/screenings/ScreeningEditPageSpec.jsx]: Opportunity for refactor? We could merge `requiredScreeningAttributes` here, which limits the number of places we have to add new data points.

File: spec/javascripts/components/screenings/WorkerSafetyCardViewSpec.jsx [hunk 9]
```diff
@@ -1,10 +1,49 @@
...
+
+  describe('when mode is set to edit', () => {
+    beforeEach(() => {
+      component = shallow(<WorkerSafetyCardView {...props} mode='edit'/>)
+    })
+    it('renders the worker safety edit card', () => {
+      expect(component.find('WorkerSafetyShowView').length).toEqual(0)
+      expect(component.find('WorkerSafetyEditView').length).toEqual(1)
+    })
+    it('is passed to onCancel', () => {
```
Comment (saraOSI) [inline spec/javascripts/components/screenings/WorkerSafetyCardViewSpec.jsx]: What is passed to `onCancel`?

File: spec/features/screening/show_screening_spec.rb (hunk 4 above)
Comment (saraOSI) [inline spec/features/screening/show_screening_spec.rb]: Also, there aren't any JS tests for the save/cancel functionality.

File: app/assets/javascripts/components/screenings/ScreeningEditPage.jsx [hunk 10]
```diff
@@ -1,4 +1,5 @@
 import * as screeningActions from 'actions/screeningActions'
+import * as screeningEditHelpers from 'utils/screeningEditHelper'
```
Comment (aaronOSI) [inline app/assets/javascripts/components/screenings/ScreeningEditPage.jsx]: `addNewAllegations` is needed in line 141 for the `AllegationsCardView`

File: spec/config/initializers/api_spec.rb [hunk 11]
```diff
@@ -3,12 +3,12 @@
 
 describe API do
   before do
```
Comment (aaronOSI) [inline spec/config/initializers/api_spec.rb]: We tried using `around` but rspec wouldn't allow stub_const to be used outside of a test context, so we eventually reverted back to the first thing we tried.

File: spec/features/screening/edit_screening_spec.rb (hunk 5 above)
Comment (aaronOSI) [inline spec/features/screening/edit_screening_spec.rb]: Added
